velox/exec/SortBuffer.cpp

```diff
@@ -285,11 +285,13 @@ void SortBuffer::spillOutput() {
 }
 
 void SortBuffer::prepareOutput(uint32_t maxOutputRows) {
-  VELOX_CHECK_GT(maxOutputRows, 0);
+  VELOX_CHECK_GT(static_cast<vector_size_t>(maxOutputRows), 0);
   VELOX_CHECK_GT(numInputRows_, numOutputRows_);
-
+  const uint64_t batchSizeMaybe = numInputRows_ - numOutputRows_;
```

Comment:
This logic is not obvious. Can we add a comment that we are trying to avoid overflow here?

I'm wondering if this is relatively common need and if we could use a helper function (+ a unit test) for this. 

CC: @Yuhta @xiaoxmeng